corehq/apps/domain/decorators.py

```diff
@@ -231,6 +232,11 @@ def login_or_token_ex(allow_cc_users=False, allow_sessions=True):
     return _login_or_challenge(tokenauth, allow_cc_users=allow_cc_users, allow_sessions=allow_sessions)
 
 
+def login_or_formplayer_ex(allow_cc_users=False, allow_sessions=True, require_user=False):
+    challenge = formplayer_as_user_auth if require_user else formplayer_auth
+    return _login_or_challenge(challenge, allow_cc_users=allow_cc_users, allow_sessions=allow_sessions)
```

Comment:
shouldn't `allow_sessions` always be `False` here? otherwise just accessing the view with a logged in user will cause the decorator to short circuit and not check the mac digest: https://github.com/dimagi/commcare-hq/blob/bbc6bb714be3bd5a512c24dd130d9c4c5e31033b/corehq/apps/domain/decorators.py#L199

Reply:
That's correct. If a session is present then it will get used for auth. This is how the restore endpoint is now so I don't want to change that. We just want formplayer to be able to auth when there is no session ID and when we don't have any user credentials (as is the case in SMS forms)